include/bluetooth/bluetooth.h

```diff
@@ -398,6 +525,140 @@ int bt_le_adv_update_data(const struct bt_data *ad, size_t ad_len,
+ *
+ *  Start advertising with a specific advertising set.
+ *  If both timeout and advertising event limit is given then the advertising
+ *  set will be stopped once one of them is reached.
+ *
+ *  @param adv    Advertising set object.
+ *  @param param  Advertise start parameters.
+ */
+int bt_le_ext_adv_start(struct bt_le_ext_adv *adv,
+			struct bt_le_ext_adv_start_param *param);
+
+/** Stop advertising with the given advertising set
+ *
+ *  Stop advertising with a specific advertising set. When using this function
+ *  the advertising sent callback will not be called.
+ *
+ *  @param adv Advertising set object.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_stop(struct bt_le_ext_adv *adv);
+
+/** Set advertising set advertising or scan response data.
+ *
+ *  Set advertisement data or scan response data. If the advertising set is
+ *  currently advertising then the advertising data will be updated in
+ *  subsequent advertising events.
+ *
+ *  If the advertising set has been configured with Primary advertising then the
+ *  maximum data length is 31 bytes. If the advertising set has been configured
+ *  for extended advertising data length, then the maximum data length defined
+ *  by the controller.
+ *
+ *  NOTE: Not all scanners support extended data length advertising data.
+ *
+ *  @param adv     Advertising set object.
+ *  @param ad      Data to be used in advertisement packets.
+ *  @param ad_len  Number of elements in ad
+ *  @param sd      Data to be used in scan response packets.
+ *  @param sd_len  Number of elements in sd
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_set_data(struct bt_le_ext_adv *adv,
+			   const struct bt_data *ad, size_t ad_len,
+			   const struct bt_data *sd, size_t sd_len);
+
+/** @brief Update advertising parameters.
+ *
+ *  Update the advertising parameters. The advertiser will have to be stopped
+ *  before the parameters can be updated.
+ *
+ *  @param adv   Advertising set object.
+ *  @param param Advertising parameters.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_update_param(struct bt_le_ext_adv *adv,
+			       const struct bt_le_adv_param *param);
+
+/** Delete advertising set.
+ *
+ *  Delete advertising set. This will free up the advertising set and make it
+ *  possible to create a new advertising set.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_delete(struct bt_le_ext_adv *adv);
+
+/** @brief Get array index of an advertising set.
+ *
+ *  This function is used to map bt_adv to index of an array of
+ *  advertising sets. The array has CONFIG_BT_MAX_ADV_SET elements.
+ *
+ *  @param adv Advertising set.
+ *
+ *  @return Index of the advertising set object.
+ *  The range of the returned value is 0..CONFIG_BT_MAX_ADV_SET-1
+ */
+u8_t bt_le_ext_adv_get_index(struct bt_le_ext_adv *adv);
```

Comment:
What exactly is the use case for this? I'd leave it out and add it later if necessary if you don't currently have a strong need for it.

Reply:
Not a strong use-case. But I find it useful in some cases, and I would like to keep it. It's implementation is quite simple as well.